shared/src/ledger/eth_bridge/bridge_pool_vp.rs

```diff
@@ -169,36 +209,62 @@ where
             return Ok(false);
         }
 
-        // check that gas fees were put into escrow
-
-        // check that the correct amount was deducted from the fee payer
-        if let Some(SignedAmount::Negative(amount)) =
-            self.account_balance_delta(&transfer.gas_fee.payer)
-        {
-            if amount != transfer.gas_fee.amount {
-                return Ok(false);
-            }
+        // if we are going to mint wNam on Ethereum, the appropriate
+        // amount of Nam must be escrowed in the Ethereum bridge VP's storage.
+        // TODO: We should look this address up from storage
+        if transfer.transfer.asset == wnam() {
+            // check that correct amount of Nam was put into escrow.
+            return if transfer.gas_fee.payer == transfer.transfer.sender {
+                if !self.check_nam_escrowed(
+                    &transfer.gas_fee.payer,
+                    &BRIDGE_POOL_ADDRESS,
+                    transfer.gas_fee.amount + transfer.transfer.amount,
+                    transfer.gas_fee.amount,
+                )? || !self.check_nam_escrowed(
+                    &transfer.transfer.sender,
+                    &Address::Internal(InternalAddress::EthBridge),
+                    transfer.gas_fee.amount + transfer.transfer.amount,
+                    transfer.transfer.amount,
+                )? {
+                    Ok(false)
+                } else {
+                    tracing::info!(
+                        "The Ethereum bridge pool VP accepted the transfer \
+                         {:?}.",
+                        transfer
+                    );
+                    Ok(true)
+                }
+            } else if !self.check_nam_escrowed(
+                &transfer.gas_fee.payer,
+                &BRIDGE_POOL_ADDRESS,
+                transfer.gas_fee.amount,
+                transfer.gas_fee.amount,
+            )? || !self.check_nam_escrowed(
+                &transfer.transfer.sender,
+                &Address::Internal(InternalAddress::EthBridge),
+                transfer.transfer.amount,
+                transfer.transfer.amount,
+            )? {
+                Ok(false)
+            } else {
+                tracing::info!(
+                    "The Ethereum bridge pool VP accepted the transfer {:?}.",
+                    transfer
+                );
+                Ok(true)
+            };
```

Comment:
I found this code a bit confusing to read, took me a while to see what was happening; maybe it was because of the repetition? it's fine for now, but I think it should be refactored later

Reply:
Nope, I'll solve it now.